processing/src/main/java/io/druid/segment/DoubleDimensionMergerV9.java

```diff
@@ -24,76 +24,22 @@
 import io.druid.segment.serde.ColumnPartSerde;
 import io.druid.segment.writeout.SegmentWriteOutMedium;
 
-import java.io.IOException;
-import java.nio.IntBuffer;
-import java.util.List;
-
-public class DoubleDimensionMergerV9 implements DimensionMergerV9<Double>
+public class DoubleDimensionMergerV9 extends NumericDimensionMergerV9
 {
-  protected String dimensionName;
-  protected final IndexSpec indexSpec;
-  private GenericColumnSerializer serializer;
-
-  public DoubleDimensionMergerV9(
-      String dimensionName,
-      IndexSpec indexSpec,
-      SegmentWriteOutMedium segmentWriteOutMedium
-  )
-  {
-    this.dimensionName = dimensionName;
-    this.indexSpec = indexSpec;
-
-    try {
-      setupEncodedValueWriter(segmentWriteOutMedium);
-    }
-    catch (IOException ioe) {
-      throw new RuntimeException(ioe);
-    }
-  }
-
-  private void setupEncodedValueWriter(SegmentWriteOutMedium segmentWriteOutMedium) throws IOException
-  {
-    this.serializer = IndexMergerV9.createDoubleColumnSerializer(
-        segmentWriteOutMedium,
-        dimensionName,
-        indexSpec
-    );
-
-    serializer.open();
-  }
-
-  @Override
-  public void writeMergedValueMetadata(List<IndexableAdapter> adapters) throws IOException
-  {
-    // double columns do not have additional metadata
-  }
-
-  @Override
-  public Double convertSegmentRowValuesToMergedRowValues(Double segmentRow, int segmentIndexNumber)
-  {
-    return segmentRow;
-  }
 
-  @Override
-  public void processMergedRow(Double rowValues) throws IOException
-  {
-    serializer.serialize(rowValues);
-  }
-
-  @Override
-  public void writeIndexes(List<IntBuffer> segmentRowNumConversions) throws IOException
+  DoubleDimensionMergerV9(String dimensionName, IndexSpec indexSpec, SegmentWriteOutMedium segmentWriteOutMedium)
   {
-    // double columns do not have indexes
+    super(dimensionName, indexSpec, segmentWriteOutMedium);
   }
 
   @Override
-  public boolean canSkip()
+  GenericColumnSerializer setupEncodedValueWriter()
   {
-    return false;
+    return IndexMergerV9.createDoubleColumnSerializer(segmentWriteOutMedium, dimensionName, indexSpec);
```

Comment:
I don't have a definite opinion. Your point makes sense, but also it makes sense to keep those methods (for Double, Float and Long) together. I would not change this in this PR, because it's not in the scope of this PR. I didn't put those methods in `IndexMergerV9` in this PR.

Reply:
Ok.